modules/mysql/main.tf

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  default_tier                            = "db-n1-standard-1"
+  default_activation_policy               = "ALWAYS"           // ON_DEMAND
+  default_crash_safe_replication          = true
+  default_disk_autoresize                 = true
+  default_disk_size                       = 10
+  default_disk_type                       = "PD_SSD"
+  default_pricing_plan                    = "PER_USE"
+  default_maintenance_window_day          = 1
+  default_maintenance_window_hour         = 23
+  default_replication_type                = "SYNCHRONOUS"
+  default_maintenance_window_update_track = "canary"
+
+  default_user_host = "%"
+
+  primary_zone       = "${lookup(var.master, "zone")}"
+  read_replica_zones = ["${compact(split(",", lookup(var.read_replica, "zones", "")))}"]
+
+  zone_mapping = {
+    enabled  = ["${local.read_replica_zones}"]
+    disabled = "${list(local.primary_zone)}"
+  }
+
+  zones_enabled = "${length(local.read_replica_zones) > 0}"
+  mod_by        = "${local.zones_enabled ? length(local.read_replica_zones) : 1}"
+
+  zones = "${local.zone_mapping["${local.zones_enabled ? "enabled" : "disabled"}"]}"
+}
+
+resource "google_sql_database_instance" "master" {
+  project          = "${var.project_id}"
+  name             = "${var.name}"
+  database_version = "${var.database_version}"
+  region           = "${var.region}"
+
+  settings {
+    tier                        = "${lookup(var.master, "tier", local.default_tier)}"
```

Comment:
To improve readability, but now we have some `master_*` arguments and current situation is odd, so I agreed about split fields into each arguments. But the `master_` prefix is not required because the `master` instance is now provided as `default`.
See: d86bddbf9ff3c732d16897e2c596ea5572f47910